src/main/java/com/google/codeu/servlets/ChartServlet.java

```diff
@@ -0,0 +1,58 @@
+package com.google.codeu.servlets;
+
+
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import java.util.Scanner;
+
+
+@WebServlet("/ChartServlet")
+public class ChartServlet extends HttpServlet {
+
+  private JsonArray charityDonationArray;
+
+  // This class could be its own file if we needed it outside this servlet
+  private static class charityDonation {
+    String charityName;
+    double donation;
+
+    private charityDonation (String charityName, double donation) {
+      this.charityName = charityName;
+      this.donation = donation;
+    }
+  }
+
+
+  @Override
+  public void init() {
+	charityDonationArray = new JsonArray();
+    Gson gson = new Gson();
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/charity-form.csv"));
+    scanner.nextLine(); //skips first line (the csv header)
+    while(scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      String curName = cells[1];
+      //System.out.println(curName);
+      double curDonation = Double.parseDouble(cells[2]);
+      //System.out.println(curDonation);
+      charityDonationArray.add(gson.toJsonTree(new charityDonation(curName, curDonation)));
+    }
+    scanner.close();
+  }
+
```

Comment:
One small change that I suggest is the indentation in line 34, and now that you have tested the code I think you can delete the print statements on line 43 and 45.

Reply:
Thanks Pardis! I forgot to delete it!

Reply:
Indentation in line 34 seems like a format problem on github. It looks fine on my computer. Could you review last commit I did just now as well? I deleted the print code. Thanks!